### PR TITLE
UF testing tools project. First tool: a way to mock Event<Type> in cdi beans

### PR DESCRIPTION
**Pull request by ederign**

### CODE REVIEW

File: uberfire-testing-utils/src/main/java/org/uberfire/event/EventSourceMock.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public Event<T> select( Annotation... qualifiers ) {
+        throw new RuntimeException( "mocking testing class" );
+    }
+
+    @Override
+    public <U extends T> Event<U> select( Class<U> subtype,
+                                          Annotation... qualifiers ) {
+        throw new RuntimeException( "mocking testing class" );
```
Comment (porcelli) [inline uberfire-testing-utils/src/main/java/org/uberfire/event/EventSourceMock.java]: UnsupportedOperationException ?

**Comment by manstis** [comment]
It'll be interesting to see how the Observer methods can be implemented without CDI. Arguably testing the Observers of Events is more important (OK, OK, as important!) than checking the firing of Events.

**Comment by ederign** [comment]
I agree, but we are talking here about unit testing. So in that case if you are testing class X and want to verify how it reacts by a event, just call the observer method in X.

But for integration tests is another topic ;)

File: uberfire-testing-utils/src/test/java/org/uberfire/mocks/EventSourceMockTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class EventSourceMockTest {
```
Comment (porcelli) [inline uberfire-testing-utils/src/test/java/org/uberfire/mocks/EventSourceMockTest.java]: duplicated?

Comment (ederign) [inline uberfire-testing-utils/src/test/java/org/uberfire/mocks/EventSourceMockTest.java]: Fixed.

**Comment by porcelli** [comment]
https://github.com/uberfire/uberfire/commit/af077fcadd6f3bacd4fd0e636cae20309931429f

**Pull request closed without merging**